### PR TITLE
PLAT-500: add constructor deduplication test before result was sent

### PR DESCRIPTION
**Pull request by bigbes**

### CODE REVIEW

File: ledger-core/virtual/integration/deduplication_test.go [hunk 1]
```diff
@@ -145,3 +147,86 @@ func TestDeduplication_Constructor_DuringExecution(t *testing.T) {
...
+
+		select {
+		case <-server.Journal.WaitAllAsyncCallsDone():
+		case <-time.After(10 * time.Second):
+			require.FailNow(t, "timeout")
+		}
+	}
+
+	{
+		assert.Equal(t, 1, typedChecker.VCallResult.Count())
```
[the remark this answers is not in the document]
Comment (bigbes) [reply ledger-core/virtual/integration/deduplication_test.go:229]: Not yet, until we'll get VFindCallResponse.

File: ledger-core/virtual/integration/deduplication_test.go [hunk 2]
```diff
@@ -145,3 +147,86 @@ func TestDeduplication_Constructor_DuringExecution(t *testing.T) {
...
+	}
+
+	synchronizeExecution := NewSynchronizationPoint(1)
+
+	{
+		requestResult := requestresult.New([]byte("123"), gen.UniqueReference())
+		requestResult.SetActivate(gen.UniqueReference(), class, []byte("234"))
+
+		executionMock := runnerMock.AddExecutionMock(calculateOutgoing(pl).String())
+		executionMock.AddStart(nil, &executionupdate.ContractExecutionStateUpdate{
```
[the remark this answers is not in the document]
Comment (bigbes) [reply ledger-core/virtual/integration/deduplication_test.go:184]: It should be checked in `mc.Finish()`, that was lost somehow.

File: ledger-core/virtual/integration/deduplication_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bigbes) [reply ledger-core/virtual/integration/deduplication_test.go:229]: Where we'll find result of that call? We don't store them.

File: ledger-core/virtual/integration/deduplication_test.go [hunk 3]
```diff
@@ -73,7 +77,7 @@ func TestDeduplication_Constructor_DuringExecution(t *testing.T) {
 		})
 	}
 
-	typedChecker := server.PublisherMock.SetTypedChecker(ctx, t, server)
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
 	typedChecker.VCallResult.SetResend(false)
 	typedChecker.VDelegatedCallRequest.SetResend(true)
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deduplication_test.go:82]: you can ignore resend here and create a delegation token by yourself to reduce test scope.

File: ledger-core/virtual/integration/deduplication_test.go [hunk 4]
```diff
@@ -99,9 +103,227 @@ func TestDeduplication_Constructor_DuringExecution(t *testing.T) {
 		assert.Equal(t, 1, typedChecker.VCallResult.Count())
 		assert.Equal(t, 1, typedChecker.VDelegatedCallRequest.Count())
 		assert.Equal(t, 1, typedChecker.VDelegatedCallResponse.Count())
-		assert.Equal(t, 1, typedChecker.VDelegatedCallRequest.Count())
+		assert.Equal(t, 1, typedChecker.VDelegatedRequestFinished.Count())
 		assert.Equal(t, 1, typedChecker.VStateReport.Count())
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deduplication_test.go:107]: why don't you add a check for VCallRequest?

File: ledger-core/virtual/integration/deduplication_test.go [hunk 5]
```diff
@@ -99,9 +103,227 @@ func TestDeduplication_Constructor_DuringExecution(t *testing.T) {
...
+		executionMock.AddStart(func(ctx execution.Context) {
+			synchronizeExecution.Synchronize()
+		}, &execution.Update{
+			Type:   execution.Done,
+			Result: requestResult,
+		})
+	}
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VCallResult.SetResend(false).ExpectedCount(100)
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/deduplication_test.go:234]: expected 100, why?

File: ledger-core/virtual/integration/deduplication_test.go [hunk 6]
```diff
@@ -32,14 +32,18 @@ import (
 )
 
 func TestDeduplication_Constructor_DuringExecution(t *testing.T) {
-	t.Log("C4998")
+	t.Log("С5206")
+
+	var (
+		mc = minimock.NewController(t)
+	)
```
Comment (MariaVasilenko) [inline ledger-core/virtual/integration/deduplication_test.go:39]: why you used var() ? this block is not hide

File: ledger-core/virtual/integration/deduplication_test.go [hunk 7]
```diff
@@ -99,9 +103,227 @@ func TestDeduplication_Constructor_DuringExecution(t *testing.T) {
...
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VCallResult.Set(func(result *payload.VCallResult) bool {
+		synchronizeExecution.Synchronize()
+		return false
+	})
+	typedChecker.VStateReport.SetResend(true)
+
+	{
+		server.SendPayload(ctx, &pl)
+	}
```
Comment (MariaVasilenko) [inline ledger-core/virtual/integration/deduplication_test.go:167]: what is the lifehack with {}?

Comment (bigbes) [reply ledger-core/virtual/integration/deduplication_test.go:167]: Just to split constructions that will modify internal state and checks

File: ledger-core/virtual/integration/deduplication_test.go (hunk 5 above)
Comment (bigbes) [reply ledger-core/virtual/integration/deduplication_test.go:234]: sorry, typo; was checking that feature with "ExpectedCount" works

**Pull request merged**